src/app/child-dev-project/progress-dashboard-widget/progress-dashboard/progress-dashboard.component.ts

```diff
@@ -29,10 +33,8 @@ export class ProgressDashboardComponent
   ngOnInit() {
     this.data = new ProgressDashboardConfig(this.dashboardConfigId);
     this.entityMapper
-      .load<ProgressDashboardConfig>(
-        ProgressDashboardConfig,
-        this.dashboardConfigId
-      )
+      .load(ProgressDashboardConfig, this.dashboardConfigId)
+      .catch(() => this.retryAfterSync())
```

Comment:
individual widgets really shouldn't have to worry about sync status. This has to move to a different layer - probably the `DashboardComponent` could trigger a full reload of widgets? Why did this come up in relation to the permission feature?

Reply:
Now the demo data is only generated **after** the login happens and therefore this would immediately create a new dashboard config and the demo data generation then creates a conflict